src/popup/components/AddressBook/AddressBookList.vue

```diff
@@ -86,15 +100,41 @@ export default defineComponent({
       }
     });
     const scrollWrapper = computed(() => scrollWrapperEl.value?.$el);
+    const protocolName = computed(() => {
+      if (protocolFilter.value) {
+        return ProtocolAdapterFactory.getAdapter(protocolFilter.value)?.protocolName;
+      }
+      return '';
+    });
+    const hideOuterButtons = computed(() => isScrolled.value || !!searchQuery.value);
+    const isSearchVisible = computed(() => (
+      (!protocolFilter.value || props.isSelector || showBookmarked.value)
+      && Object.keys(addressBookFilteredByProtocol.value).length > 9
+    ) || hideOuterButtons.value);
+    const hasBookmarkedEntries = computed(
+      () => addressBookFilteredByProtocol.value.some((entry) => entry.isBookmarked),
+    );
+
+    function handleAddressBookItemClick(entry: IAddressBookEntry) {
+      if (!entry) {
+        return;
+      }
+      if (props.isSelector) {
+        emit('selectAddress', entry.address);
+      } else {
+        router.push({ name: ROUTE_ADDRESS_BOOK_EDIT, params: { id: entry.address } });
+      }
```

Comment:
As I mentioned in the `isSelector` prop definition: this logic should happen on the parent level.